packages/next/src/server/dev/hot-reloader.ts

```diff
@@ -413,6 +419,55 @@ export default class HotReloader {
       )
   }
 
+  private async getVersionInfo(span: Span) {
+    const versionInfoSpan = span.traceChild('get-version-info')
+    return versionInfoSpan.traceAsyncFn<VersionInfo>(async () => {
+      try {
+        const installed = require('next/package.json').version
+        const installedParsed = semver.parse(installed)
+
+        const res = await fetch(
+          'https://registry.npmjs.org/-/package/next/dist-tags'
+        )
+
+        if (!res.ok) return { installed, staleness: 'unknown' }
+
+        const tags = await res.json()
+        const latest = semver.parse(tags.latest)
+        const canary = semver.parse(tags.canary)
+
+        if (installedParsed && latest && canary) {
```

Comment:
Lets add unit tests for this code to make sure it behaves correctly with all these conditions